packages/admin/src/appHeader/dropdown/AppHeaderDropdown.tsx

```diff
@@ -13,6 +13,7 @@ interface Props extends AppHeaderButtonProps {
     buttonChildren?: React.ReactNode;
     dropdownArrow?: ((isOpen: boolean) => React.ReactNode) | null;
     popoverProps?: Partial<PopoverProps>;
+    customShowContentState?: [boolean, React.Dispatch<React.SetStateAction<boolean>>];
```

Comment:
strange API :)

better would be:
```
showContent: boolean,
onShowContentChange: (show: boolean) => void
```

see also
https://github.com/vivid-planet/comet-admin/blob/next/packages/admin/src/common/buttons/split/SplitButton.tsx#L36

Reply:
instead of "showContent" you could also use the term "open" or "visible". I think mui dialog uses open.

Reply:
> strange API :)

that's React's useState-api :)

The idea is to have an uncontrolled component by default, which manages its own state.
By passing  a `customShowContentState`, which is exactly the return value of a `useState<boolean>()`, the component can be made controlled (state lives outside of the component). (I call it "Hybrid Component")

This api would mean the component is only uncontrolled:

```
showContent: boolean,
onShowContentChange: (show: boolean) => void
```

Alternative might be this:

```
showContent?: boolean,
onShowContentChange?: (show: boolean) => void
```

but the user should not be able to pass only `showContent` and not `onShowContentChange`, or vice versa, it must be either fully controlled or fully uncontrolled.


We tried out this pattern here: https://gitlab.vivid-planet.com/vivid/comet/-/blob/master/admin/cms-admin/src/common/DropdownMenuItem.tsx#L31



> instead of "showContent" you could also use the term "open" or "visible". I think mui dialog uses open.

I agree,

the name could be improved to `customOpenState`, or maybe `controlOpenState`, or `controlledOpenState` to be more accurate

Reply:
> Alternative might be this:
>>  showContent?: boolean,
>> onShowContentChange?: (show: boolean) => void

yes of course optional, my mistake.

----

I don't see an advantage of exposing the useState API. Your argument that you ensure both props are set is true, but that is exactly the same pattern for any controlled native `input`. Devs are familar with that.

the pattern is used here
https://github.com/vivid-planet/comet-admin/blob/next/packages/admin/src/common/buttons/split/SplitButton.tsx#L11

Reply:
The pattern in SplitBox is great too! I just thought the "useStatePattern" is already used in some of the components.

Reply:
I changed it to use the pattern from `SplitButton`